library/L2_HAL/displays/lcd/test/st7066u_test.cpp

```diff
@@ -0,0 +1,343 @@
+        static_cast<uint8_t>(Value(kDefaultConfiguration) | Value(bus_mode) |
+                             Value(kDisplayMode) | Value(kFontStyle)),
+        Value(St7066u::Command::kTurnDisplayOn),
+        Value(St7066u::Command::kClearDisplay),
+      };
+      kEightBitBusLcd.Initialize();
+    }
+
+    Verify(Method(mock_rs, SetDirection).Using(Gpio::Direction::kOutput),
+           Method(mock_rw, SetDirection).Using(Gpio::Direction::kOutput),
+           Method(mock_e, SetDirection).Using(Gpio::Direction::kOutput),
+           Method(mock_e, Set).Using(Gpio::kHigh),
+           Method(mock_data_bus, Initialize),
+           Method(mock_data_bus, SetDirection).Using(Gpio::Direction::kOutput));
+  }
+
+  SECTION("Write Nibble")
+  {
+    constexpr uint8_t kExpectedNibble = 0b1010;
+    St7066u lcd                       = kFourBitBusLcd;
+
+    lcd.WriteNibble(St7066u::WriteOperation::kCommand, kExpectedNibble);
+    Verify(Method(mock_rs, Set)
+               .Using(Gpio::State(St7066u::WriteOperation::kCommand)),
+           Method(mock_rw, Set).Using(Gpio::State::kLow),
+           Method(mock_data_bus, Write).Using(kExpectedNibble),
+           Method(mock_e, Set).Using(Gpio::State::kLow),
+           Method(mock_e, Set).Using(Gpio::State::kHigh));
+
+    lcd.WriteNibble(St7066u::WriteOperation::kData, kExpectedNibble);
+    Verify(
+        Method(mock_rs, Set).Using(Gpio::State(St7066u::WriteOperation::kData)),
+        Method(mock_rw, Set).Using(Gpio::State::kLow),
+        Method(mock_data_bus, Write).Using(kExpectedNibble),
+        Method(mock_e, Set).Using(Gpio::State::kLow),
+        Method(mock_e, Set).Using(Gpio::State::kHigh));
+  }
+
+  SECTION("WriteByte")
+  {
+    constexpr uint8_t kExpectedByte = 0b1010'1010;
+    St7066u lcd                     = kEightBitBusLcd;
+
+    lcd.WriteNibble(St7066u::WriteOperation::kCommand, kExpectedByte);
+    Verify(Method(mock_rs, Set)
+               .Using(Gpio::State(St7066u::WriteOperation::kCommand)),
+           Method(mock_rw, Set).Using(Gpio::State::kLow),
+           Method(mock_data_bus, Write).Using(kExpectedByte),
+           Method(mock_e, Set).Using(Gpio::State::kLow),
+           Method(mock_e, Set).Using(Gpio::State::kHigh));
+
+    lcd.WriteNibble(St7066u::WriteOperation::kData, kExpectedByte);
+    Verify(
+        Method(mock_rs, Set).Using(Gpio::State(St7066u::WriteOperation::kData)),
+        Method(mock_rw, Set).Using(Gpio::State::kLow),
+        Method(mock_data_bus, Write).Using(kExpectedByte),
+        Method(mock_e, Set).Using(Gpio::State::kLow),
+        Method(mock_e, Set).Using(Gpio::State::kHigh));
+  }
+
+  SECTION("Write Command using 4-bit Bus")
+  {
+    constexpr Gpio::State kExpectedRegisterSelect =
+        Gpio::State(St7066u::WriteOperation::kCommand);
+    uint8_t expected_data;
+    St7066u lcd = kFourBitBusLcd;
+
+    SECTION("ClearDisplay")
+    {
+      expected_data = static_cast<uint8_t>(St7066u::Command::kClearDisplay);
```

Comment:
```suggestion
      expected_data = Value(St7066u::Command::kClearDisplay);
```
You can use `Value` here.